### ISSUE TITLE
False positive for type incompatibility with circular dependencies

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#3561]
This is follow up for #3542

Here's the example:

https://github.com/last-partizan/pyright-examples/tree/1f632bbb0579a3f82e67f8ddb8c7bc8d854444f0

**Describe the bug**
Clone repo, `pip install django django-types`, run pyright `1.1.253`, it fails with following message:

```
/home/serg/src/tests/test-pyright/example/models.py
  /home/serg/src/tests/test-pyright/example/models.py:3:6 - error: Import "django.db" could not be resolved (reportMissingImports)
1 error, 0 warnings, 0 informations
```

As in previous bug, order in which files scanned matters. You should run `pyright` for all directory, not single file.

**Comment by erictraut** [comment]
Are you sure you're running pyright 1.1.253? If you run `pyright --version`, which version string is printed?

It looks like you're using the Python wrapper (from `pypi`) to install pyright. Have you set the environment variable `PYRIGHT_PYTHON_FORCE_VERSION` to "latest" so it always uses the latest published version?

**Comment by erictraut** [comment]
I suspect that you were running an older version of pyright because I'm not able to repro the error you reported above. However, I did find another unfortunate side effect of my previous fix that forced me to revert it and implement a different solution.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.254, which I just published. It will also be included in the next release of pylance.
